scripts/run.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+var fs = require('fs');
+var os = require('os');
+var path = require('path');
+var webpack = require('webpack');
+
+var paths = require('../config/paths');
+
+// Hacky command-line parsing. All flags are just ignored and there
+// may be no args.
+// TODO: support passing args to the script we're running
+var nonflags = process.argv.filter(function(x) { return x[0] !== '-'; });
+if (nonflags.length != 3) {
+  console.error('Usage: react-scripts run foo.js');
+  console.error(
+    'foo.js is your script, resolved relative to the src directory.')
+  process.exit(1);
+}
+var scriptName = nonflags[2];
+var pathToEntry = path.join(paths.appSrc, scriptName);
+
+// Figure out what all the node_modules are.
+// We don't want to compile the node_modules, but we still want to
+// do a commonjs require for them.
+var nodeModules = {};
+fs.readdirSync(paths.appNodeModules).filter(function(x) {
```

Comment:
This won't cover eg `require("lodash/fp/extend")` (it'll get bundled instead of marked as an external).
If you want, you could just use [webpack-node-externals](https://www.npmjs.com/package/webpack-node-externals) to handle this part.